Replace osm_rawdata with raw_data_api_py adding highway category

Sujanadh opened the pull request:
## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation
- [X] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue
Issues:
- #2628
- #2485
- #2588

## Describe this PR

- Replaced `osm_rawdata` with `raw_data_api_py`
- Fixed functionality to generate highway data
- Used `St_Within`, With setting it false I am getting entire polygon of country from the `AOI` (not sure how)
- Kept `St_Within` false for highways so all the highways crossing the `AOI` will be generated
- Added `geometryType` in the parameter; without setting `geometryType` I wasn't able to generate highway using `raw_data_api_py`
- Our enum `POLYLINE` doesn't align with` raw-data-api` where `LINE` is used for highways as enum
-Updated xlsform to fix minor issue where users are allowed to continue filling up form without selecting existing `building` or `new_feature` 
    changes:
    - `"relevant": "(${feature_exists} = 'yes') or (${status} != '3')",` --->  `"relevant": "(${feature_exists} = 'yes') or (${status} = '3')"`


**yet to do**:
- [x] yet to release `raw-data-api-py` package @spwoodcock
- [x] update the package in field-tm
- [x] update highways xlsform
- [x] replace osm_rawdata with raw_data_api_py during osm conflation as well
- [ ] add feature to parse yaml directly

## Screenshots

**Buildings:**
![image](https://github.com/user-attachments/assets/325c8735-d211-4278-a7c3-ab9657c6c128)

**Highways:**
![image](https://github.com/user-attachments/assets/29046497-6207-48c7-9ae4-73cf27160f7f)

**Centroids:**
![image](https://github.com/user-attachments/assets/4adcb4f4-8657-45a6-be75-015871ea3eff)


## Checklist before requesting a review

- 📖 Read the Field-TM Contributing Guide: <https://github.com/hotosm/field-tm/blob/main/CONTRIBUTING.md>
- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

## [optional] What gif best describes this PR or how it makes you feel?

spwoodcock commented:
I publishing and added the raw-data-api-py package to the PR 👍

Sujanadh commented:
I have fixed osm conflation as well and need to update the package itself since we don't have anything that gives us plain geojson. Additionally, there was an issue with the function `javarosa_to_geojson` it was giving me invalid coordinates with an empty list at the end of the coordinates.

spwoodcock commented:
> I have fixed osm conflation as well and need to update the package itself since we don't have anything that gives us plain geojson. Additionally, there was an issue with the function `javarosa_to_geojson` it was giving me invalid coordinates with an empty list at the end of the coordinates. 

Sorry I'm a bit confused by this - does accessing raw-data-api need conflation anywhere?

Sujanadh commented:
We have `get_osm_geometries` function in `postgis_utils` which is used during conflation, where we get osm features in plain geojson using the task boundary, and we conflate each feature with submission. It is using `osm_rawdata` so we need to replace it with `raw_data_api_py` . So we need to add a function in `raw_data_api_py` to return plain geojson with a separate snapshot.

spwoodcock commented:
Merged, released, and updated to 0.3.0 here 😄

Sujanadh commented:
## Updates:
- I have converted all the yaml data models into json data models using `osm_rawdata`
- Made relevant changes using json model
- Fixed pytest

Frontend `e2e-tests` are still failing.
Except that we are good to go. @spwoodcock You can merge at any time :+1:.

spwoodcock reviewed (approved):
Huge PR - really great job! 🙌
